projects/core/components/primitive-checkbox/primitive-checkbox.component.ts

```diff
@@ -27,6 +27,10 @@ export class TuiPrimitiveCheckboxComponent {
     @tuiDefaultProp()
     disabled = false;
 
+    @Input()
+    @tuiDefaultProp()
+    readOnly = false;
```

Comment:
`PrimitiveCheckbox` is already readonly, no need for additional input. Maybe just add `_readonly` class to it in `Checkbox` component.

Reply:
fixed